superset/assets/src/query/datasourceBuilder.ts

```diff
@@ -0,0 +1,19 @@
+import { FormData } from './formData';
+
+enum DatasourceType {
+  Table = 'table',
+  Druid = 'druid',
+}
+
+export interface Datasource {
+  id: number;
+  type: DatasourceType;
+}
+
+export default function build(formData: FormData): Datasource {
```

Comment:
could we match the name the file for easier debugging? I want to add the lint rule for this.

Reply:
Yep! so `datasourceBuilder` as the function name? Do we care for a verb like `buildDatasource`? Preference? I don't have a strong opinion about this. 😄

Reply:
If it is a single function, I would prefer verb.